### ISSUE TITLE
[QUESTION] The whole history sequence predict, is there way that much faster that historical_forecast

### ISSUE THREAD

**Issue by ruixu2** [unit8co/darts#2790]
I tried to history forecast and compute metrics including mae mape and r2 score.

But as mentiond #2780  , it was too too slow using backtest or historical_forecast functions and i can not find the reason.

So is there other ways to finish the history forecast.

- In pytorch's deep learning model, the torch_metrics can input and training with the metric validation.
- But in other models,  

Deep learning train, 8 iter/s
![Image](https://github.com/user-attachments/assets/68e0e093-f591-44d3-aadf-84bcbec43a78)

backtest is about  #1iter/s

**Comment by dennisbader** [comment]
Let's continue this discussion in #2780. I'll close this one to avoid duplicates.
